### PR TITLE
Merge release/jdbc/1.0.0 back to main 

### PR DESCRIPTION
**Pull request by saragluna**
# Description

The `azure-identity-extensions` 1.0.0 has been GAed, this PR will merge the release branch back to main. 

# All SDK Contribution checklist:
- [x] **The pull request does not introduce [breaking changes]**
- [x] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [x] **I have read the [contribution guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md).**

## [General Guidelines and Best Practices](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#developer-guide)
- [x] Title of the pull request is clear and informative.
- [x] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### [Testing Guidelines](https://github.com/Azure/azure-sdk-for-java/blob/main/CONTRIBUTING.md#building-and-unit-testing)
- [x] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: eng/versioning/version_client.txt [hunk 1]
```diff
@@ -400,8 +398,7 @@ com.azure.tools:azure-sdk-build-tool;1.0.0-beta.1;1.0.0-beta.2
 # In the pom, the version update tag after the version should name the unreleased package and the dependency version:
 # <!-- {x-version-update;unreleased_com.azure:azure-core;dependency} -->
 unreleased_com.azure:azure-core;1.35.0-beta.1
-unreleased_com.azure:azure-identity-providers-jdbc-mysql;1.0.0
-unreleased_com.azure:azure-identity-providers-jdbc-postgresql;1.0.0
+unreleased_com.azure:azure-identity-extensions;1.0.0
```
Comment (backwind1233) [inline eng/versioning/version_client.txt:401]: Do we still use this now?

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml [hunk 2]
```diff
@@ -74,14 +73,8 @@
...
-        <artifactId>azure-identity-providers-jdbc-mysql</artifactId>
-        <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-identity-providers-jdbc-mysql;dependency} -->
-      </dependency>
-
-      <dependency>
-        <groupId>com.azure</groupId>
-        <artifactId>azure-identity-providers-jdbc-postgresql</artifactId>
-        <version>1.0.0-beta.1</version> <!-- {x-version-update;com.azure:azure-identity-providers-jdbc-postgresql;dependency} -->
+        <artifactId>azure-identity-extensions</artifactId>
+        <version>1.0.0</version> <!-- {x-version-update;unreleased_com.azure:azure-identity-extensions;dependency} -->
```
Comment (backwind1233) [inline sdk/boms/spring-cloud-azure-dependencies/pom.xml:77]: since azure-identity-extensions has been released, can we use the GA version here?

**Comment by backwind1233** [comment]
@saragluna do we need another PR to [spring-boot3 ](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch?

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (saragluna) [reply eng/versioning/version_client.txt:401]: No, but I will use another PR to address this and other issues.

File: sdk/boms/spring-cloud-azure-dependencies/pom.xml (hunk 2 above)
Comment (saragluna) [reply sdk/boms/spring-cloud-azure-dependencies/pom.xml:77]: still will use another PR.

File: eng/versioning/version_client.txt (hunk 1 above)
Comment (backwind1233) [reply eng/versioning/version_client.txt:401]: Ok

**Comment by saragluna** [comment]
> @saragluna do we need another PR to [spring-boot3 ](https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3) branch?

No, these are non-spring folder changes, will treat this as an Azure SDK change.

**Comment by backwind1233** [comment]
> 

So, I think we should make it clear.

Because I think if there are changes under folder sdk/spring, we should also sync with https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3. 
But here is different from what I discussed with @moarychan , so what's exactly the rules to follow to sync both branches.

**Review by backwind1233 — APPROVED** [review]
LGTM

**Comment by saragluna** [comment]
> > 
> 
> @saragluna So, I think we should make it clear.
> 
> Because I think if there are changes under folder sdk/spring, we should also sync with https://github.com/Azure/azure-sdk-for-java/tree/feature/spring-boot-3. But here is different from what I discussed with @moarychan , so what's exactly the rules to follow to sync both branches.
> 
> image

No need to sync this whole PR, but only these two changes:
- https://github.com/Azure/azure-sdk-for-java/pull/32430/commits/10b07d94791b5793cf53a29db8791fd7837ae767
- https://github.com/Azure/azure-sdk-for-java/pull/32430/commits/61ac4f16b8936fec62ab37c50d843ea2fed11028

**Pull request merged**